tests/test_pipeline/test_hierarchical_pipeline.py

```diff
@@ -0,0 +1,279 @@
+            "target": [1, 1, 1] * 4,
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def market_level_constant_hierarchical_df():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02", "2000-01-03"] * 2,
+            "segment": ["X"] * 3 + ["Y"] * 3,
+            "target": [2, 2, 2] * 2,
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def market_level_constant_hierarchical_df_exog():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02", "2000-01-03", "2000-01-04", "2000-01-05"] * 2,
+            "segment": ["X"] * 5 + ["Y"] * 5,
+            "regressor": [1, 1, 1, 1, 1] * 2,
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def market_level_constant_hierarchical_ts(market_level_constant_hierarchical_df, hierarchical_structure):
+    ts = TSDataset(df=market_level_constant_hierarchical_df, freq="D", hierarchical_structure=hierarchical_structure)
+    return ts
+
+
+@pytest.fixture
+def market_level_constant_hierarchical_ts_w_exog(
+    market_level_constant_hierarchical_df, market_level_constant_hierarchical_df_exog, hierarchical_structure
+):
+    ts = TSDataset(
+        df=market_level_constant_hierarchical_df,
+        df_exog=market_level_constant_hierarchical_df_exog,
+        freq="D",
+        hierarchical_structure=hierarchical_structure,
+        known_future="all",
+    )
+    return ts
+
+
+@pytest.fixture
+def product_level_constant_hierarchical_ts_w_exog(
+    product_level_constant_hierarchical_df, market_level_constant_hierarchical_df_exog, hierarchical_structure
+):
+    ts = TSDataset(
+        df=product_level_constant_hierarchical_df,
+        df_exog=market_level_constant_hierarchical_df_exog,
+        freq="D",
+        hierarchical_structure=hierarchical_structure,
+        known_future="all",
+    )
+    return ts
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="a", source_level="b", period=1, method="AHP"),
+        BottomUpReconciliator(target_level="a", source_level="b"),
+    ),
+)
+def test_init_pass(reconciliator):
+    model = NaiveModel()
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=1)
+    assert isinstance(pipeline.reconciliator, type(reconciliator))
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="market", source_level="total", period=1, method="AHP"),
+        BottomUpReconciliator(target_level="total", source_level="market"),
+    ),
+)
+def test_fit_mapping_matrix(market_level_simple_hierarchical_ts, reconciliator):
+    model = NaiveModel()
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=1)
+    with patch.object(type(reconciliator), "fit", wraps=reconciliator.fit) as mock:
```

Comment:
Actually you can do sth like this for simplicity:
```
pipeline.fit = Mock()
```

Reply:
Here we want to test if reconsiliator method `fit` is called while fitting a pipeline.

Reply:
Then 
```
pipeline. reconciliator.fit = Mock()
```